### PR TITLE
Fix #1839: Fixed overlapping progressbar and hint bulb 

### LINKED ISSUE #1839: Question player progress bar & hint bulb overlap unexpectedly
Question player progress bar & hint bulb overlap unexpectedly. See screenshot:

Ideally, we should make sure there's sufficient separation between the question progress bar & hint bulbs so that they don't overlap.

Note that this also seems to happen in the question player.

Final Solution as per below mentioned conversation.

**Scale Progress Bar to Fit** 

"Scale Progress Bar to Fit" will animate to the right once the hint button appears, which looks more natural flowing than if it were to move up.
As per the image above the `Progress Bar` size will change when the hint bulb appears and the `Progress Bar` size will change (going back to normal) once the hint bulb disappears. Note that this change should have animation in place so that it does not look abrupt change.

Reference: https://github.com/oppia/oppia-android/issues/1839#issuecomment-704698704

### PR DESCRIPTION
**Pull request by aishapervin**
## Explanation
Fixes #1839
i am adding a constraint from `progressbar` to` hints_and_solution_fragments_container` to solve this issue.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by aishapervin** [comment]
@rt4914 please check.

**Review by FareesHussain — APPROVED** [review]
LGTM, thanks.

**Comment by anandwana001** [comment]
Can we try one thing here: 
Write a test case with `isCompletlyDisplayed()` and see if it passes with current fix and fail without fix. 
Just a suggestion if it works out.

File: app/src/main/res/layout/question_player_fragment.xml [hunk 1]
```diff
@@ -93,7 +93,38 @@
         android:progressDrawable="@drawable/progress_bar"
         app:layout_constraintBottom_toTopOf="@id/question_progress_text"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent" />
+        app:layout_constraintStart_toEndOf="@id/hints_and_solution_fragment_container" />
+      <FrameLayout
```
Comment (rt4914) [inline app/src/main/res/layout/question_player_fragment.xml:98]: Add one blank line above this.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aishapervin Nice work. Just one nit change suggested.

File: app/src/main/res/layout/question_player_fragment.xml (hunk 1 above)
Comment (aishapervin) [reply app/src/main/res/layout/question_player_fragment.xml:98]: done.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**